src/chai/ManagedArray.inl

```diff
@@ -58,21 +58,71 @@ CHAI_HOST_DEVICE ManagedArray<T>::ManagedArray():
 {
 #if !defined(__CUDA_ARCH__)
   m_resource_manager = ArrayManager::getInstance();
+
+  m_pointer_record = new PointerRecord{};
+  m_pointer_record->m_size = 0;
+  m_pointer_record->m_user_callback = [](Action, ExecutionSpace, size_t) {};
+
+  for (int space = CPU;  space < NUM_EXECUTION_SPACES; space++) {
+    m_pointer_record->m_allocators[space] = 
+      m_resource_manager->getAllocatorId(ExecutionSpace(space));
+  }
 #endif
 }
 
 template<typename T>
 CHAI_INLINE
 CHAI_HOST_DEVICE ManagedArray<T>::ManagedArray(
-    size_t elems, ExecutionSpace space):
-  m_active_pointer(nullptr),
-  m_resource_manager(nullptr),
-  m_elems(elems),
-  m_pointer_record(nullptr)
+    std::initializer_list<chai::ExecutionSpace> spaces,
+    std::initializer_list<umpire::Allocator> allocators):
+  ManagedArray()
 {
 #if !defined(__CUDA_ARCH__)
-  m_resource_manager = ArrayManager::getInstance();
+  int i = 0;
+  for (auto& space : spaces) {
```

Comment:
@trws - looking at this method, do you think there is a better way to implement this? I could use arrays instead - do you think that would be a good choice?

Note that this method is instantiated by user code, so we should be able to infer the array size as a template parameter.

Reply:
I like taking arrays, or at least a constant-sized thing, if it only makes sense to take a known number of items.  If this is meant to have dynamic rank, then I'd leave it as is, otherwise I'd probably switch it to something that enforces the restriction at compile time.